dev-docs/bidder-adaptor.md

```diff
@@ -12,259 +11,504 @@ nav_section: adapters
+    /* Change this for your adapter as needed to provide a renderer. */
+}
+```
 
-1. Load the userSync module in the adapter file
-2. Register the sync
+## Supporting Native
 
-{% highlight js %}
-import { userSync } from 'src/userSync.js';
-
-   // ... code ...
-   if ($$PREBID_GLOBAL$$.userSync.iframeEnabled) {
-     userSync.registerSync('iframe', 'exampleAdapter', 'http://example.com/iframe-url');
-   } else {
-     userSync.registerSync('image', 'exampleAdapter', 'http://example.com/pixel');
-   }
-   // ... code ...
-{% endhighlight %}
+In order for your bidder to support the native media type, you need 2 things:
+
+1. Your (server-side) bidder needs to respond with a bid that has native information.
+2. Your (client-side) bidder adapter needs to unpack the server's bid into a Prebid-compatible bid populated with the required native information.
+
+For example, below is the [code in the AppNexus AST adapter](https://github.com/prebid/Prebid.js/blob/master/modules/appnexusAstBidAdapter.js#L192) that achieves #2.  As you can see, we:
+
+1. Check for native information on the bid response.
+2. Fill in the bid's `native` object with information from the bid.
 
-The general syntax for the userSync.registerSync() function:
 {% highlight js %}
-userSync.registerSync(type, adapterName, url);
-{% endhighlight %}
 
-Note that the userSync object contains the registerSync function. The example above shows a destructed import, which 
-eliminates the need to refer the to function through its object.
+/* Does the bidder respond with native information? */
+} else if (rtbBid.rtb.native) {
 
-Parameters for registerSync:
+  const native = rtbBid.rtb.native;
 
-{: .table .table-bordered .table-striped }
-| Param | Type | Description |
-| ----- | ---- | ----------- |
-| `type` | string enum | The type of sync, either 'image' or 'iframe'. |
-| `adapterName` | string | The name of the adapter. Used for logging. |
-| `url` | string | The fully qualified URL.|
+  /* Bidder supports native, so let's populate our Prebid-compatible
+  bid with native info */
+  bid.native = {
+    title: native.title,
+    body: native.desc,
+    cta: native.ctatext,
+    sponsoredBy: native.sponsored,
+    image: native.main_img && native.main_img.url,
+    icon: native.icon && native.icon.url,
+    clickUrl: native.link.url,
+    impressionTrackers: native.impression_trackers,
+  };
 
-{: .alert.alert-success :}
-See the [Publisher API Reference]({{site.baseurl}}/dev-docs/publisher-api-reference.html#module_pbjs.userSync) for more information about how this feature works.
+{% endhighlight %}
 
-<div class="alert alert-danger" role="alert">
-<p>
-Note that in Prebid 1.0, use of the registerSync() function will be required for all user ID activity.
-</p>
-</div>
+## Unit Tests
 
-## Helper functions
+Every adapter submission must include unit tests. See existing test suites in [test/spec/modules](https://github.com/prebid/Prebid.js/tree/master/test/spec/modules).
```

Comment:
Are there any specific requirements for unit tests that we should call out here?

Reply:
Good callout - in e80533f I added a link to the Prebid.js CONTRIBUTING.md which has a bunch of testing requirements listed.